### ISSUE TITLE
"-.5" dec> is busted

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1229]
The change to fix radix support in c4bd4dc1cc701c757411d69cb000550995dd56bc seems to have caused a regression:

```factor
{ -0.5 } [ "-.5" dec> ] unit-test
```

**Comment by mrjbq7** [comment]
This seems to fix it and all tests pass, but it seems like our math parsing is a little hokey:

```diff
diff --git a/core/math/parser/parser.factor b/core/math/parser/parser.factor
index c10ac80..d0058a7 100644
--- a/core/math/parser/parser.factor
+++ b/core/math/parser/parser.factor
@@ -274,13 +274,37 @@ DEFER: @neg-digit
         [ @pos-first-digit ]
     } case ; inline
 
-: @first-char-no-radix ( i number-parse n char -- n/f )
+: with-no-radix ( i number-parse n quot -- n/f )
+    [
+        swap {
+            { CHAR: b [ pick radix>> 16 = [ CHAR: b swap call ] [ @abort ] if ] }
+            { CHAR: o [ @abort ] }
+            { CHAR: x [ @abort ] }
+            [ swap call ]
+        } case
+    ] curry require-next-digit ; inline
+
+: @neg-first-digit-no-radix ( i number-parse n char -- n/f )
+    {
+        { CHAR: . [ ->required-mantissa ] }
+        { CHAR: 0 [ [ @neg-digit ] with-no-radix ] }
+        [ @neg-digit ]
+    } case ; inline
+
+: @pos-first-digit-no-radix ( i number-parse n char -- n/f )
     {
-        { CHAR: - [ [ @neg-digit ] require-next-digit ?neg ] }
-        { CHAR: + [ [ @pos-digit ] require-next-digit ] }
+        { CHAR: . [ ->required-mantissa ] }
+        { CHAR: 0 [ [ @pos-digit ] with-no-radix ] }
         [ @pos-digit ]
     } case ; inline
 
+: @first-char-no-radix ( i number-parse n char -- n/f )
+    {
+        { CHAR: - [ [ @neg-first-digit-no-radix ] require-next-digit ?neg ] }
+        { CHAR: + [ [ @pos-first-digit-no-radix ] require-next-digit ] }
+        [ @pos-first-digit-no-radix ]
+    } case ; inline
+
 PRIVATE>
 
 : string>number ( str -- n/f )
```

**Comment by mrjbq7** [comment]
@erg thoughts?

**Comment by mrjbq7** [comment]
Fixed in 94bb834eaef1a500e3ead50fa2ccea7685c8808f.
